compiler/rustc_middle/src/ty/parameterized.rs

```diff
@@ -0,0 +1,119 @@
+use rustc_hir::def_id::DefId;
+use rustc_index::vec::{Idx, IndexVec};
+
+use crate::middle::exported_symbols::ExportedSymbol;
+use crate::mir::Body;
+use crate::thir::abstract_const::Node;
+use crate::ty::{
+    self, Const, FnSig, GeneratorDiagnosticData, GenericPredicates, Predicate, TraitRef, Ty,
+};
+
+pub trait ParameterizedOverTcx: 'static {
```

Comment:
The question: Does this really need to be `: 'static`? These impls also could be `impl<'a> ParameterizedOverTcx for Thing<'a> { type Value<'tcx> = Thing<'tcx>; }`, right? Because  This is *just* encoding a higher-kinded `lifetime -> type`.

Reply:
Yeah, there's no good reason why we _need_ `Self: 'static`. 

I treat it more like a mnemonic, where the lifetime error you'd get is there to discourage you from constructing `Lazy<Something<'not_static>>`, since there is no guarantee that the `'not_static` lifetime is not related to what you get out of `lazy.decode(meta)`.